README.md

```diff
@@ -162,11 +167,79 @@ If we already have a WMS GetMap URL, we can use it directly:
   const imageBlob4 = await legacyGetMapFromUrl(fullUrlWithWmsQueryString, ApiType.PROCESSING);
 ```
 
-# Running examples
+# Authentication for Processing API
+
+Requests to Processing API need to be authenticated.
+Authentication is done by requesting an authentication token and setting it as described in [Layers](#layers).
+The authentication token is retrieved by making a request to the `https://services.sentinel-hub.com/oauth/token` endpoint with the OAuth Client's id and secret.
+
+To get the OAuth Client's id and secret, a new OAuth Client must be created in [**User settings**](https://apps.sentinel-hub.com/dashboard/#/account/settings) on **Sentinel Hub Dashboard** under **OAuth clients**.
+OAuth Client's secret is shown only before the creation process is finished so be mindful to save it.
+
+In javascript requesting the authentication token can be done with builtin XMLHttpRequest:
+
+```javascript
+const clientId = /* OAuth Client's id, best to put it in .env file and use it from there */;
+const clientSecret = /* OAuth client's secret, best to put it in .env file and use it from there */;
+
+let xhr = new XMLHttpRequest();
+xhr.open("POST", 'https://services.sentinel-hub.com/oauth/token', true);
+xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+xhr.onreadystatechange = function() {
+  if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+    console.log(xhr.responseText);
```

Comment:
I would suggest calling `setAuthToken` here instead of calling `console.log`, as this is what most (all?) of the users will do.

Reply:
True, changed.